src/hypofuzz/docs/manual/collection.rst

```diff
@@ -1,18 +1,43 @@
 Test collection
 ===============
 
-HypoFuzz uses :pypi:`pytest` to collect the test functions to fuzz, with almost identical command-line interfaces.  If you're using ``pytest ...`` to run your tests, ``hypothesis fuzz -- ...`` will fuzz them.
+Before HypoFuzz starts fuzzing by assigning tests to workers, it first needs to know what tests are available. HypoFuzz does so by running a collection step.
 
-This also means you can use the standard pytest ``-k`` selector to configure which tests to run, with ``hypothesis fuzz -- -k ...``. See pytest's docs on :ref:`pytest:select-tests` and :ref:`pytest:test discovery` for more details.
+How HypoFuzz collects tests
+---------------------------
 
-Note that tests which use `pytest fixtures <https://docs.pytest.org/en/stable/fixture.html>`__ are not collected, as they may behave differently outside of the pytest runtime. We recommend using a context manager and the ``with`` statement instead. Support for pytest fixtures is on :doc:`our roadmap </roadmap>`.
+HypoFuzz uses :pypi:`pytest` to collect test functions, with an almost identical command-line interface.
+
+If you're using ``pytest ...`` to run your tests, ``hypothesis fuzz -- ...`` will fuzz them. Everything after ``--`` is passed to a call to ``pytest --collect-only`` in HypoFuzz.
+
+Among others, this means you can use the standard pytest ``-k`` selector to configure which tests to run, with ``hypothesis fuzz -- -k ...``. See pytest's docs on :ref:`pytest:select-tests` and :ref:`pytest:test discovery` for more details.
+
+Concretely, if your source layout looks like this:
+
+.. code-block::
+
+    src/
+        ...
+    tests/
+        test_a.py
+        test_b.py
+
+and you normally run your tests with ``pytest``, then:
+
+* ``hypothesis fuzz`` will fuzz all your tests
+* ``hypothesis fuzz -- tests/test_a.py`` will fuzz the tests in ``test_a.py``
+* ``hypothesis fuzz -- -k selector_string`` will fuzz the tests matching ``selector_string`` (see :ref:`pytest:select-tests`).
+
+Pytest plugins during collection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``hypothesis fuzz`` does not run pytest plugins during collection. If your standard pytest command includes plugin parameters like ``pytest -n auto`` (from e.g. :pypi:`pytest-xdist`), do not pass these parameters to ``hypothesis fuzz``.
 
-Support for other test runners, such as :mod:`python:unittest`, is on our roadmap.
 
 Skipping tests under HypoFuzz
 -----------------------------
 
-If you would like to skip a test under HypoFuzz while keeping it part of your normal test suite, you can use |in_hypofuzz_run|:
+If you would like to skip a test under HypoFuzz while keeping it part of your normal test suite, you can use :ref:`pytest:pytest.mark.skipif ref` in combination with |in_hypofuzz_run|:
```

Comment:
Code block below uses that as a boolean when it's now a function!

I'd also suggest showing the sys.modules trick, to avoid importing hypofuzz if it's not present, and document that under in_hypofuzz_run too.

Reply:
I'd hope they don't need the sys.modules trick, since importing hypofuzz won't have side-effects on `in_hypofuzz_run`, so they can do so unconditionally!

Reply:
sure, but that's just a generic perf comment, not hypofuzz-specific. I might mention it briefly but I wouldn't go into too much detail